### PR TITLE
Remove GDMPointer.Value

### PR DESCRIPTION
**Pull request by hazzik**
Replace `GDMPointer.Value` with `GDMPointer.GetPtrValue<T>` where possible. 

Next big challenge in my view is `GDMTag.Clear` method. I think it needs to be removed.

### CODE REVIEW

File: projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs [hunk 1]
```diff
@@ -41,8 +41,7 @@ public GDMAssociation Association
             set {
                 if (fAssociation != value) {
                     fAssociation = value;
-                    //fTempPerson = fBase.Context.Tree.GetPtrValue(fAssociation);
-                    fTempPerson = fAssociation.Individual;
+                    fTempPerson = fBase.Context.Tree.GetPtrValue(fAssociation);
```
Comment (hazzik) [inline projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs:44]: I'm not sure why you have not replaced this by yourself

File: projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs [hunk 2]
```diff
@@ -63,8 +62,7 @@ public override bool Accept()
                 }
 
                 fAssociation.Relation = rel;
-                fAssociation.Individual = fTempPerson;
-                //fBase.Context.Tree.SetPtrValue(fAssociation, fTempPerson);
+                fBase.Context.Tree.SetPtrValue(fAssociation, fTempPerson);
```
Comment (hazzik) [inline projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs:65]: I'm not sure why you have not replaced this by yourself

File: projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs (hunk 1 above)
Comment (Serg-Norseman) [reply projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs:44]: There was a nasty bug in the test of this part. I didn't manage (something I don't understand) using NSubstitute functions. When I make a stub from BaseWindow, I needed Returns (realInstanceOfGDMTree) to be executed at the BaseContext level - and this did not.

**Review by Serg-Norseman — APPROVED** [review]
Excellent! Very neat work, thank you! 👍

File: projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs (hunk 1 above)
Comment (Serg-Norseman) [reply projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs:44]: Bug with using NSubstitute in the /projects/GKTests/GKCore/Controllers/AssociationEditControllerTests.cs/Test_AssociationEditDlgController()

File: projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs (hunk 2 above)
Comment (Serg-Norseman) [reply projects/GKCore/GKCore/Controllers/AssociationEditDlgController.cs:65]: Bug with using NSubstitute in the /projects/GKTests/GKCore/Controllers/AssociationEditControllerTests.cs/Test_AssociationEditDlgController()

**Pull request merged**